### ISSUE TITLE
Publishing library that uses Apollo Kotlin 4.0.0 to maven repo requires declaring explicit dependency on certain Gradle task

### ISSUE THREAD

**Issue by seab4ss** [apollographql/apollo-kotlin#6146]
### Version

4.0.0

### Summary

Our library project includes Apollo Kotlin. We use `maven-publish` plugin for publishing the library `.aar` file. After migrating from `v3.8.2` to `v4.0.0`, we started seeing error when attempting to publish our library `.aar` artifact. It seems that task `mylibrary:sourceReleaseJar` has to depend on `:mylibrary:generateMylibrarysvcApolloSources` task
To fix the issue,  i added the following code in `afterEvaluate {}` in the `myLibrary/build.gradle.kts`:
```
afterEvaluate {
    val apolloSrcTask = project.tasks.getByName(“generateMylibrarysvcApolloSources”)
    val srcReleaseTask = project.tasks.getByName("sourceReleaseJar")
    srcReleaseTask.dependsOn(apolloSrcTask)
}
```
Is there some configuration setting  in Apollo Gradle Plugin  that will fix this issue?

### Steps to reproduce the behavior

```
dependencies {

    // @Keep annotation
    implementation(libs.androidx.annotation)

    // Coroutines
    implementation(libs.kotlinx.coroutines.core)
    implementation(libs.kotlinx.coroutines.android)
    testImplementation(libs.kotlinx.coroutines.test)

    // Apollo core runtime dependencies
    implementation(libs.apollo.kotlin.runtime)
...
}


apollo {
    service("Mylibrarysvc) {
        // apollo compiler by default will generate Kotlin models from queries, mutations etc
        packageName.set("mylibrary")
        // Keep schemaFile
        schemaFile.set(file("mylibraryFilepath"))
        // explicitly set srcDir
        srcDir(file("graphqlFilesFilepath"))
}
}

```

here is publishing section in `mylibrary/build.gradle.kts`
```
publishing {
    publications {
        register<MavenPublication>(ARTIFACT_NAME) {
            groupId = ARTIFACT_PACKAGE
            artifactId = ARTIFACT_NAME
            version = getPublishingVersion(ARTIFACT_VERSION)

            // Library documentation created with Dokka
            artifact(dokkaDocsJarTask)

            // Note that when running the Maven Publish Plugin with AGP,
            // the software components are not created directly when the plugin is applied.
            // They are instead created during the afterEvaluate() callback step.
            // Therefore, the publication that selects the software component
            // must also be configured during the afterEvaluate() step.
            afterEvaluate {
                from(components["release"])
            }
        }
    }

    repositories {

        mavenLocal()
    }
}
```

run  `myLibrary/publishToMavenLocal` task in Android Studio

### Logs

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ‘:mylibrary:sourceReleaseJar' (type 'SourceJarTask').
  - Gradle detected a problem with the following location: '/Users/Myself/mylibrary/build/generated/source/apollo/mylibrarysvc'.
    
    Reason: Task ':mylibrary:sourceReleaseJar' uses this output of task ':mylibrary:generateMylibrarysvcApolloSources' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':mylibrary:generateMylibrarysvcApolloSources' as an input of ':mylibrary:sourceReleaseJar'.
      2. Declare an explicit dependency on ':mylibrary:generatemylibrarysvcApolloSources' from ':mylibrary:sourceReleaseJar' using Task#dependsOn.
      3. Declare an explicit dependency on ':mylibrary:generatemylibrarysvcApolloSources' from ':mylibrary:sourceReleaseJar' using Task#mustRunAfter.
    
    For more information, please refer to https://docs.gradle.org/8.2/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.

**Comment by martinbonnin** [comment]
Hi 👋 I'm guessing you're using something like so to configure the android publication?

```kotlin
android {
  publishing {
    singleVariant("release") {
        withSourcesJar()
    }
  }
}
```

The generated sources are registered using [srcDir()](https://github.com/apollographql/apollo-kotlin/blob/2d1d25fd28ecdbd24ae0cc8cfef7ffb618f03f35/libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo/gradle/internal/AndroidPluginFacade.kt#L70) and I would have expected that to carry task dependencies but maybe there's more to it. I'll take a deeper look. In the meantime your workaround should work fine.

**Comment by seab4ss** [comment]
Yes. that's how we configured the android publication. Thank you

**Comment by martinbonnin** [comment]
@seab4ss fix is [here](https://github.com/apollographql/apollo-kotlin/pull/6149). There's a surprising amount of context and backstory but I think/hope it's good. If you get a chance to try it out in the [SNAPSHOTs](https://www.apollographql.com/docs/kotlin/#snapshots), let me know how it goes (should land in the SNAPSHOTs once merged and deployed, hopefully today).

**Comment by martinbonnin** [comment]
(PR merged, SNAPSHOTs are deploying [here](https://github.com/apollographql/apollo-kotlin/actions/runs/10881586406))

**Comment by seab4ss** [comment]
@martinbonnin this issue is fixed in Apollo Kotlin `v4.0.1`. Thank you!

**Comment by martinbonnin** [comment]
Glad to hear that! Thanks for the follow up!
